### ISSUE TITLE
Cannot start oncall as a k8s container

### ISSUE THREAD

**Issue by macetw** [grafana/oncall#1353]
I tried using kubernetes to launch grafana/oncall as a pod, and it fails to start with django during the initialization. 

Here is the log as shown from kubernetes:

```
2023-02-20 23:37:42 uWSGI http bound on 0.0.0.0:8080 fd 7
2023-02-20 23:37:42 uwsgi socket 0 bound to TCP address 127.0.0.1:41635 (port auto-assigned) fd 6
2023-02-20 23:37:42 Python version: 3.9.16 (main, Jan 23 2023, 23:56:42)  [GCC 8.3.0]
2023-02-20 23:37:42 *** Python threads support is disabled. You can enable it with --enable-threads ***
2023-02-20 23:37:42 Python main interpreter initialized at 0x56342f9384e0
2023-02-20 23:37:42 your server socket listen backlog is limited to 100 connections
2023-02-20 23:37:42 your mercy for graceful operations on workers is 60 seconds
2023-02-20 23:37:42 mapped 855210 bytes (835 KB) for 5 cores
2023-02-20 23:37:42 *** Operational MODE: preforking ***
2023-02-20 23:37:42 Traceback (most recent call last):
2023-02-20 23:37:42   File "/usr/local/lib/python3.9/site-packages/django/utils/connection.py", line 58, in __getitem__
2023-02-20 23:37:42     return getattr(self._connections, alias)
2023-02-20 23:37:42   File "/usr/local/lib/python3.9/site-packages/asgiref/local.py", line 105, in __getattr__
2023-02-20 23:37:42     raise AttributeError(f"{self!r} object has no attribute {key!r}")
2023-02-20 23:37:42 AttributeError: <asgiref.local.Local object at 0x7fe3389603d0> object has no attribute 'default'
2023-02-20 23:37:42 
During handling of the above exception, another exception occurred:

2023-02-20 23:37:42 Traceback (most recent call last):
2023-02-20 23:37:42   File "/etc/app/./engine/__init__.py", line 5, in <module>
2023-02-20 23:37:42     from .celery import app as celery_app
2023-02-20 23:37:42   File "/etc/app/./engine/celery.py", line 24, in <module>
2023-02-20 23:37:42     connection.cursor()
2023-02-20 23:37:42   File "/usr/local/lib/python3.9/site-packages/django/utils/connection.py", line 15, in __getattr__
2023-02-20 23:37:42     return getattr(self._connections[self._alias], item)
2023-02-20 23:37:42   File "/usr/local/lib/python3.9/site-packages/django/utils/connection.py", line 60, in __getitem__
2023-02-20 23:37:42     if alias not in self.settings:
2023-02-20 23:37:42   File "/usr/local/lib/python3.9/site-packages/django/utils/functional.py", line 48, in __get__
2023-02-20 23:37:42     res = instance.__dict__[self.name] = self.func(instance)
2023-02-20 23:37:42   File "/usr/local/lib/python3.9/site-packages/django/utils/connection.py", line 45, in settings
2023-02-20 23:37:42     self._settings = self.configure_settings(self._settings)
2023-02-20 23:37:42   File "/usr/local/lib/python3.9/site-packages/django/db/utils.py", line 144, in configure_settings
2023-02-20 23:37:42     databases = super().configure_settings(databases)
2023-02-20 23:37:42   File "/usr/local/lib/python3.9/site-packages/django/utils/connection.py", line 50, in configure_settings
2023-02-20 23:37:42     settings = getattr(django_settings, self.settings_name)
2023-02-20 23:37:42   File "/usr/local/lib/python3.9/site-packages/django/conf/__init__.py", line 82, in __getattr__
2023-02-20 23:37:42     self._setup(name)
2023-02-20 23:37:42   File "/usr/local/lib/python3.9/site-packages/django/conf/__init__.py", line 69, in _setup
2023-02-20 23:37:42     self._wrapped = Settings(settings_module)
2023-02-20 23:37:42   File "/usr/local/lib/python3.9/site-packages/django/conf/__init__.py", line 170, in __init__
2023-02-20 23:37:42     mod = importlib.import_module(self.SETTINGS_MODULE)
2023-02-20 23:37:42   File "/usr/local/lib/python3.9/importlib/__init__.py", line 127, in import_module
2023-02-20 23:37:42     return _bootstrap._gcd_import(name[level:], package, level)
2023-02-20 23:37:42 ModuleNotFoundError: No module named 'settings.prod'
2023-02-20 23:37:42 unable to load app 0 (mountpoint='') (callable not found or import error)
2023-02-20 23:37:42 *** no app loaded. going in full dynamic mode ***
2023-02-20 23:37:42 *** uWSGI is running in multiple interpreter mode ***
2023-02-20 23:37:42 spawned uWSGI master process (pid: 1)
2023-02-20 23:37:42 spawned uWSGI worker 1 (pid: 7, cores: 1)
2023-02-20 23:37:42 spawned uWSGI worker 2 (pid: 8, cores: 1)
2023-02-20 23:37:42 spawned uWSGI worker 3 (pid: 9, cores: 1)
2023-02-20 23:37:42 spawned uWSGI worker 4 (pid: 10, cores: 1)
2023-02-20 23:37:42 spawned uWSGI worker 5 (pid: 11, cores: 1)
2023-02-20 23:37:42 spawned uWSGI http 1 (pid: 12)
```
The following is my pod, keeping it super basic:
```
apiVersion: v1
kind: Pod
metadata:
  name: grafana-oncall
spec:
  containers:
  - name: grafana-oncall
    image: grafana/oncall:v1.1.21
    ports:
    - containerPort: 8080
      name: grafanaoncall
      protocol: TCP
```

**Comment by macetw** [comment]
This looked like it could be helpful, but not to me as a user.
https://stackoverflow.com/questions/7367577/django-shell-no-module-named-settings

**Comment by Davenury** [comment]
Have you tried using helm chart for that? As far as I know there are multiple env variables that are injected to the pod maybe you lack some configuration that is included there?

**Comment by macetw** [comment]
I did not use a helm chart. My experience with helm is limited, and I failed when I tried following the steps:

_(I just went with the "example.com" in the suggested install step)_

```
$ helm install \
>     --wait \
>     --set base_url=example.com \
>     --set grafana."grafana\.ini".server.domain=example.com \
>     release-oncall \
>     grafana/oncall
Error: INSTALLATION FAILED: unable to build kubernetes objects from release manifest: unable to recognize "": no matches for kind "PodSecurityPolicy" in version "policy/v1beta1"
```

**Comment by Davenury** [comment]
I got around the problem you've described by setting env variable `DJANGO_SETTINGS_MODULE` to `setting.hobby` (as far as I noticed, there's also `settings/helm` but I don't know the full scope of options). There are other problems though.

As far as the helm installation goes, I suspect you have too high k8s version - [PodSecurityPolicy is not supported in 1.26](https://kubernetes.io/docs/concepts/security/pod-security-policy/) - I don't know your case though - are you trying to play around with oncall (I would recommend docker-compose then), deploy it to production environment (helm is still the best way imo but probably doesn't work in 1.26 - didn't test it though) or just pointing out a bug (with that I will not be able to help, as I'm not Grafana employee :sweat_smile:)?
